### PR TITLE
docs: Remove ENV `ONE_DIR`

### PR DESCRIPTION
**Pull request by polarathene**
# Description

I don't see this as a breaking change given the logic we have for it.

- I have brought this up a few times in the past, this PR finally contributes the suggestion to remove `ONE_DIR`.
- It doesn't really serve a purpose anymore, the `/var/mail-state` location existing is the real detection for triggering the runtime-state consolidation feature.
- I've updated all docs referencing it, and the first commit improves the documentation on our volumes (_I can split this out into a separate PR if preferred, although the `ONE_DIR` changes in docs relates to centralizing the docs for our DMS state volume_).

I've explained DMS Volumes several times in past issues ([example](https://github.com/docker-mailserver/docker-mailserver/issues/3753#issuecomment-1879919623)), so that should be a helpful addition to reference to link users to instead 👍 

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] New and existing unit tests pass locally with my changes
- [x] **I have added information about changes made in this PR to `CHANGELOG.md`**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,9 @@ The most noteworthy change of this release is the update of the container's base
 
 ### Updates
 
+- **Environment Variables:**
+  - `ONE_DIR` has been removed (legacy ENV).
```
Comment (polarathene) [inline CHANGELOG.md:39]:
```suggestion
  - `ONE_DIR` has been removed (legacy ENV) ([#3840](https://github.com/docker-mailserver/docker-mailserver/pull/3840))
```

**Comment by polarathene** [comment]
Just some thoughts on the lint failure as I didn't expect this (_feel free to ignore this comment_):
- It's only because I modified the last line in VSCode, while other files saved didn't adjust it.
- This contributes to a prior confusion about "final newline" being a line-ending only, but in VSCodes case the default is probably where I got the idea that it required a full blank line 😓 

---

image

```console
# 0 for missing final newline, 1 for present:
$ tail -c 1 docs/content/config/environment.md | wc -l
0
```

We already have that prevented by [a recent PR for user config parsing](https://github.com/docker-mailserver/docker-mailserver/pull/3819) (_where the [last report was related to VSCode defaults](https://github.com/docker-mailserver/docker-mailserver/issues/3546#issuecomment-1907617250)_), and the [ECLint check config](https://github.com/docker-mailserver/docker-mailserver/blob/0c7e49e654fb64a0b23abae2b423f54e8312beeb/test/linting/lint.sh#L19-L34) catches it here in PRs.

---

Perhaps a git pre-commit hook could better prevent it (_I haven't checked if that's implicit, or requires the contributor to [use an editor plugin like for `.editorconfig`](https://github.com/docker-mailserver/docker-mailserver/blob/v13.3.1/.editorconfig#L13)_), there doesn't seem to be an equivalent [`.gitattributes` feature like we do for preventing CRLF line-endings](https://github.com/docker-mailserver/docker-mailserver/pull/3527).

image

I'm wondering if there's any value in highlighting that default setting for VSCode in our docs (_with [required configuration change](https://stackoverflow.com/questions/50162952/vs-code-end-of-file-new-line-not-persisting/50330619#50330619)_) 🤔

VSCode otherwise expects a blank line (_a separate setting enabled by default optionally also renders it as a separate blank line visually_):

image

**Comment by georglauterbach** [comment]
I have `Insert Final Newline` set to true, but I didn't know about `Render Final Newline` (which is set to `dimmed` for me by default).

**Review by georglauterbach — APPROVED** [review]
I like the first commit in particular; it always bothered me how we only document the volumes in the FAQ 👍🏼

**Comment by polarathene** [comment]
I'll leave this a little bit longer for @casperklein to possibly chime in since I recall `ONE_DIR` discussions with him in the past too.

**Comment by casperklein** [comment]
Short and sweet:

- Let's remove ONE_DIR 👍 
- I haven't enough time to think this completely through. If there are no drawbacks for people, currently still using ONE_DIR, then it's fine to merge. Otherwise I would wait for the next major release.

---

I am also using this in VSCode:

```
    "files.trimTrailingWhitespace": true,
    "files.insertFinalNewline": true,
```

**Review by casperklein — APPROVED** [review]

**Pull request merged**